core/src/main/java/org/jdbi/v3/BuiltInArgumentFactory.java

```diff
@@ -36,14 +41,14 @@
  * class directly, but instead should bind your object with the
  * {@link SQLStatement} convenience methods.
  */
-public class BuiltInArgumentFactory<T> implements ArgumentFactory<T> {
+public class BuiltInArgumentFactory implements ArgumentFactory<Object> {
```

Comment:
To clarify: If the class is going to have a `T` parameter, than for the life of a particular generic object, that `T` should conceptually point to a single type. In this case we were using it as an end-run to save on casts.